modulus/models/diffusion/song_unet.py

```diff
@@ -149,6 +149,8 @@ def __init__(
         encoder_type: str = "standard",
         decoder_type: str = "standard",
         resample_filter: List[int] = [1, 1],
+        gridtype: str = "sinusoidal",
+        task: str = "regression",
```

Comment:
We should keep these network architectures as generic as possible. Instead of passing the task arg, consider directly passing the pos_embd.